sabnzbd/bpsmeter.py

```diff
@@ -243,6 +244,25 @@ def update(self, server=None, amount=0):
             self.bps = (self.bps * (self.last_update - self.start_time) + amount) / (t - self.start_time)
         except:
             self.bps = 0.0
+            self.server_bps = {}
+
+        if server:
+            try:
+                self.server_bps[server] = (
+                    self.server_bps.get(server, 0.0) * (self.last_update - self.start_time) + amount
+                ) / (t - self.start_time)
+            except:
+                self.server_bps[server] = 0.0
+
+        for s in self.server_bps:
+            if s == server:
+                continue
+            try:
+                self.server_bps[s] = (self.server_bps.get(s, 0.0) * (self.last_update - self.start_time) + 0) / (
+                    t - self.start_time
+                )
+            except:
+                self.server_bps[s] = 0.0
```

Comment:
It feels like we are doing the same thing here twice, I suggest could combine:
```python
#        if server:
#            try:
#                self.server_bps[server] = (
#                    self.server_bps.get(server, 0.0) * (self.last_update - self.start_time) + amount
#                ) / (t - self.start_time)
#            except:
#                self.server_bps[server] = 0.0

        for server_to_update in self.server_bps:
            try:
                self.server_bps[server_to_update] = (self.server_bps.get(server_to_update, 0.0) * (self.last_update - self.start_time) + amount * (server_to_update == server) / (
                    t - self.start_time
                )
            except:
                self.server_bps[server_to_update] = 0.0
```
The `amount * (server_to_update == server)` is a bit not-so-nice-code-style, but it saves us making an extra if.

Reply:
This change alone doesn't work because then the server won't be added to self.server_bps, so it will always be empty.

The + 0 is just silly, I should have removed it after debugging.